tests/test_api.py

```diff
@@ -725,6 +726,22 @@ def test_targetfile_from_data(self) -> None:
         targetfile_from_data = TargetFile.from_data(target_file_path, data)
         targetfile_from_data.verify_length_and_hashes(data)
 
+    def test_metafile_from_data(self) -> None:
+         data = b"Inline test content"
+
+         # Test with a valid hash algorithm
+         metafile = MetaFile.from_data(
+             1, data, ["sha256"]
+         )
+         metafile.verify_length_and_hashes(data)
```

Comment:
I think this does not actually test that the length and hashes are really used  (as the method by design succeeds even if length and hashes are not set) so that might be useful to verify.

Maybe an additional assert that e.g. `metafile.verify_length_and_hashes(b"")` fails? 

(No need to do anything more extensive as `test_length_and_hash_validation()` is extensive: this would just confirm that length and hashes are there after `MetaFile.from_data()`)

Reply:
the case that isn't theoretically covered is if (now or in a future change), the Metafile actually did not store the length and hashes: then the call just succeeds regardless of the content.

My original suggestion was a sanity check to ensure the provided length/hashes are actually used in verify_length_and_hashes:
```python
# assert that provided length is actually used in verify 
with self.assertRaises(LengthOrHashMismatchError):
    metafile.verify_length_and_hashes(b"")
```

**But** the other option (that might actually be more readable in the test) would be to compare the metafile you produced to one made by the other constructor

```python
self. assertEqual(
    metafile,
    MetaFile(1, 19, {'sha256': 'fcee2e6d56ab08eab279016f7db7e4e1d172ccea78e15f4cf8bd939991a418fa'})
)
```

I think the latter one would make a lot of sense sense and should be understandable. What do you think?

Reply:
I was about to ...  but then noticed the failing docs build (see other comment). please go ahead and add the test as well if it's not too much trouble